src/middleware.ts

```diff
@@ -13,10 +13,15 @@ export function createMiddleware(
     const serverOrigin = `${req.protocol}://${req.get('host')}`
 
     // Ensure the request body input passed to the MockedRequest
-    // is always a string. Custom middleware like "express.json()"
-    // may coerce "req.body" to be an Object.
+    // is always a buffer. Custom middleware like "express.json()"
+    // may alread have parsed "req.body". If no middleware is used,
+    // respect the original buffer
     const requestBody =
-      typeof req.body === 'string' ? req.body : JSON.stringify(req.body)
+      req.body instanceof Uint8Array
```

Comment:
This looks great!

Please, could you also add a test for this behavior? Take a look at the [existing tests](https://github.com/mswjs/http-middleware/tree/9f91ae28e2ce1026819278808436762adeef7596/test) for inspiration. 

This scenario would come down to:

1. Prepare the setup, have no `express.json()` middleware.
2. Perform a request.
3. Assert the correct request body.

1. Prepare the setup, have the `express.json()` middleware.
2. Perform a request.
3. Assert that the body is still correct.

We can put these two in a single test file to reflect on this behavior.

Reply:
yeah, sure thing!

Reply:
After setting up this test, I discovered that the root cause of my issue was that I was using `express.raw()` instead of `express.json()` (as opposed to _no_ middleware, as I originally assumed) as a workaround for https://github.com/mswjs/http-middleware/issues/17.

In the end, _some_ body parsing middleware is necessary, so maybe this change is unnecessary. However, I still think this change is worthwhile, as using `express.raw()` is technically more performant since it doesn't have to parse/unparse.

I also had to make the test setup a bit more complicated since the open-draft test-server automatically applies `bodyParser.json()` middleware and there is no way to disable it.